app/lib/search/backend.dart

```diff
@@ -143,6 +279,19 @@ class SearchBackend {
       ...?pv.pubspec!.topics,
     ].join(' ');
 
+    // select the latest entity updated timestamp (when available)
+    final sourceUpdated = [
+      p.updated,
+      scoreCard?.updated,
+    ].fold<DateTime?>(
+      null,
+      (p, v) {
+        if (v == null) return p;
+        if (p == null) return v;
+        return v.isAfter(p) ? v : p;
+      },
+    );
```

Comment:
@isoos consider `whereNotNull()` and maybe even `max`, see:
https://pub.dev/documentation/collection/latest/collection/IterableComparableExtension.html


```suggestion
    // select the latest entity updated timestamp (when available)
    final sourceUpdated = [
      p.updated,
      scoreCard?.updated,
    ].whereNotNull().max;
```